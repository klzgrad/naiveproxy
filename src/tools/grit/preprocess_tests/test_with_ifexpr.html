<!doctype html>
<html>
<head>
<meta charset="utf-8">
</head>
<body>
Always present text
<if expr="foo">
Text for foo only
</if>
<if expr="orange">
Text for orange only
</if>
More text for everyone
</body>
</html>
