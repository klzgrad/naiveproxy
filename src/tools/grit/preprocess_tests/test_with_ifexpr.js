// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

Polymer({
  is: 'cr-foo',

  _template: html`
    <if expr="bar">
      <button on-click="onClick_">I should be included in HTML</button>
    </if>
    <if expr="apple">
      <div>I should be excluded from HTML</div>
    </if>
  `,

  onClick_() {
    // <if expr="orange">
    console.log('I should be excluded from JS');
    console.log('I also should be excluded');
    // </if>
    // <if expr="foo">
    console.log('I should be included in JS');
    // </if>
  }
});
