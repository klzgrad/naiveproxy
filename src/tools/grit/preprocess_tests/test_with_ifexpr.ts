// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

export function clickHandler() {
  // <if expr="orange">
  console.log('I should be excluded from TS');
  console.log('I also should be excluded');
  // </if>
  // <if expr="foo">
  console.log('I should be included in TS');
  // </if>
}
