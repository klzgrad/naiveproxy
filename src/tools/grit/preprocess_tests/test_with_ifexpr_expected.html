<!doctype html>
<html>
<head>
<meta charset="utf-8">
</head>
<body>
Always present text

Text for foo only

<!--grit-removed-lines:2-->
More text for everyone
</body>
</html>
